src/librustc_codegen_llvm/consts.rs

```diff
@@ -437,7 +440,18 @@ impl StaticMethods for CodegenCx<'ll, 'tcx> {
                 //
                 // We could remove this hack whenever we decide to drop macOS 10.10 support.
                 if self.tcx.sess.target.target.options.is_like_osx {
-                    let sect_name = if alloc.bytes.iter().all(|b| *b == 0) {
+                    let all_bytes_range = 0..alloc.len();
+                    let is_zeroed = alloc.relocations.len() == 0 && {
+                        // Treats undefined bytes as if they were defined with the byte value that
+                        // happens to be currently assigned in mir. This is valid since reading
+                        // undef bytes may yield arbitrary values.
+                        //
+                        // FIXME: ignore undef bytes even with representation `!= 0`.
+                        alloc.raw_bytes_with_undef_and_ptr(all_bytes_range)
```

Comment:
```suggestion
                        //
                        // The `raw` method is okay here because we checked relocations, and because we are doing
                        // this access to inspect the interpreter state (not as part of the interpreter execution).
                        alloc.raw_bytes_with_undef_and_ptr(all_bytes_range)
```